src/cljs/ladders/core.cljs

```diff
@@ -2,7 +2,27 @@
     (:require [reagent.core :as reagent :refer [atom]]
               [reagent.session :as session]
               [secretary.core :as secretary :include-macros true]
-              [accountant.core :as accountant]))
+              [accountant.core :as accountant]
+              [cljs.pprint :as pprint]))
+
+(defn assoc-if [pred map key val]
+  (if (pred)
+    (assoc map key val)
+    map))
+
+(defn menu [selected]
+  (let [menu-options [["/" "Home"]
+                      ["/users" "Users"]
+                      ["/result" "Add Challenge"]]]
+    [:ul {:class "nav nav-pills"}
+     (map (fn [[url option-name]]
+            (let [li-attrs {:role "presentation"}
```

Comment:
```
let [class (if (= selection option-name) "active" "")
     li-attrs {:role "presentation" :class class}]
```

this is the only use of assoc-if so far and I think it complicates the function.